subprojects/base-services/src/main/java/org/gradle/internal/work/DefaultWorkerLeaseService.java

```diff
@@ -101,6 +101,17 @@ public DefaultWorkerLease getWorkerLease() {
     }
 
     @Override
+    public void withSharedLease(WorkerLease sharedLease, Runnable action) {
```

Comment:
done, I copied/adapted two existing tests for that, but had them use `withSharedLease`.